### PR TITLE
fix(elb): allow inverting load balancer task count metric statistics

### LINKED ISSUE #392: Support for recommended statistics for ELB metrics
### Feature scope

ELB

### Describe your suggested feature

To monitor network load balancer metrics, it is recommended to track the maximum `HealthyHostCount` and the minimum `UnHealthyHostCount` as per the guidelines at
<https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-cloudwatch-metrics.html#metric-statistics>.

At the moment, we can only track the minimum value of `HealthyHostCount` and the maximum value of `UnHealthyHostCount`. It would be great if we could also have a way to monitor the recommended statistics.

### PR DESCRIPTION
**Pull request by jrchen-amzn**
# Summary

To monitor network load balancer metrics, track the maximum `HealthyHostCount` and the minimum `UnHealthyHostCount` as per the guidelines at
<https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-cloudwatch-metrics.html#metric-statistics>.

Fixes #392

# Tests
```
yarn build
```
The `package:js` build step has succeeded.

---

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license_

### CODE REVIEW

**Comment by krnwsn** [comment]
Hi,

I hope you are doing well. I have tested it and it works fine on my end. I would appreciate it if you could review it and merge it as soon as possible. This pull request is important for me because it solves a problem I was facing with this project. 

If you need any clarification or have any feedback on this pull request, please don’t hesitate to comment and let me know.

Thank you for your time and attention. 😊
krnwsn

**Comment by krnwsn** [comment]
I appreciate your response and your insight on the breaking change. I also think it would be better to preserve the old behavior. I will make the necessary changes to the PR accordingly.

**Comment by krnwsn** [comment]
Hi,

I'm having trouble finding a simple way to retain the old behaviour. I think I need to add the related prop(like `invertStatisticsOfTaskCountEnabled`) to several interfaces like `BaseLoadBalancedFargateServiceMonitoringProps`, `NetworkLoadBalancerMetricFactoryProps`, `ApplicationLoadBalancerMetricFactoryProps` and so on. Is there a better way to do this change?

I also opened an issue(<https://github.com/cdklabs/cdk-monitoring-constructs/issues/392>) for this feature request. I appreciate your advice and feedback. Thank you for your time and attention!

**Comment by krnwsn** [comment]
Hello, I have made the changes you suggested in my pull request. Could you review it and let me know if it meets your expectations? I am grateful for your time and feedback!

**Comment by krnwsn** [comment]
Hello @echeung-amzn , I would greatly appreciate it if you could take a moment to take a look at it and let me know if there are any issues or questions. I value your support and consideration. Thank you very much!

File: lib/monitoring/aws-ecs-patterns/Ec2ServiceMonitoring.ts [hunk 1]
```diff
@@ -102,6 +102,19 @@ interface BaseLoadBalancedEc2ServiceMonitoringProps
...
+   *
+   * When `invertLoadBalancerStatisticsOfTaskCountEnabled` is set to false, the minimum of `HealthyHostCount` and the maximum of `UnHealthyHostCount` are monitored.
+   * When `invertLoadBalancerStatisticsOfTaskCountEnabled` is set to true, the maximum of `HealthyHostCount` and the minimum of `UnHealthyHostCount` are monitored.
+   *
+   * `invertLoadBalancerStatisticsOfTaskCountEnabled` is recommended to set to true as per the guidelines at
+https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-cloudwatch-metrics.html#metric-statistics
+   *
+   * @default false
+   */
+  readonly invertLoadBalancerStatisticsOfTaskCountEnabled?: boolean;
```
Comment (echeung-amzn) [inline lib/monitoring/aws-ecs-patterns/Ec2ServiceMonitoring.ts:117]: Maybe `invertLoadBalancerTaskCountMetricsStatistics`?

**Review by echeung-amzn — COMMENTED** [review]
Generally LGTM, just a minor naming suggestion.

File: lib/monitoring/aws-ecs-patterns/Ec2ServiceMonitoring.ts (hunk 1 above)
Comment (krnwsn) [reply lib/monitoring/aws-ecs-patterns/Ec2ServiceMonitoring.ts]: Thank you for your review of my pull request, and I agree that `invertLoadBalancerTaskCountMetricsStatistics` is more appropriate. I will make the necessary changes and update the pull request as soon as possible. Thank you again for your feedback and guidance.

**Comment by krnwsn** [comment]
I have completed the updates to the pull request. I would be grateful if you could review it at your earliest convenience. Please let me know if there are any further issues or if you require any additional information. Thank you for taking the time to review the changes:)

**Review by echeung-amzn — APPROVED** [review]

**Pull request merged**